Sparkle/SUUpdateValidator.m

```diff
@@ -91,23 +84,23 @@ - (BOOL)validateWithUpdateDirectory:(NSString *)updateDirectory
 
     NSURL *installSourceURL = [NSURL fileURLWithPath:installSource];
 
-    if (!prevalidatedDsaSignature) {
+    if (!self.prevalidatedSignature) {
         // Check to see if we have a package or bundle to validate
         if (isPackage) {
-            // If we get here, then the appcast installation type was lying to us.. This error will be caught later when starting the installer.
             // For package type updates, all we do is check if the DSA signature is valid
-            BOOL validationCheckSuccess = [SUDSAVerifier validatePath:downloadPath withEncodedDSASignature:DSASignature withPublicDSAKey:host.publicDSAKey];
+            BOOL validationCheckSuccess = [SUSignatureVerifier validatePath:downloadPath withSignatures:signatures withPublicKeys:publicKeys];
             if (!validationCheckSuccess) {
                 SULog(SULogLevelError, @"DSA signature validation of the package failed. The update contains an installer package, and valid DSA signatures are mandatory for all installer packages. The update will be rejected. Sign the installer with a valid DSA key or use an .app bundle update instead.");
             }
             return validationCheckSuccess;
         } else {
             // For application bundle updates, we check both the DSA and Apple code signing signatures
-            return [self validateBundleUpdateForHost:host newBundleURL:installSourceURL archivePath:downloadPath DSASignature:DSASignature];
+            return [self validateUpdateForHost:host downloadedToPath:downloadPath newBundleURL:installSourceURL signatures:signatures];
         }
     } else if (isPackage) {
-        // We already prevalidated the package and nothing else needs to be done
-        return YES;
+        // We shouldn't get here because we don't validate packages before extracting them currently
+        SULog(SULogLevelError, @"Error: not expecting to find package after being required to validate update before extraction");
+        return NO;
```

Comment:
This is incorrect for 2.x which pre-validates archives of pkg's. 2.x currently requires adding an installationType tag to the appcast if a pkg is used (for also another auth related reason); the installer later verifies that. So the code that was removed here and the above comment ("If we get here, then the appcast installation type was lying to us.") should be reverted.

Reply:
I'm sure you're correct, so let me go work through the logic myself and then I'll remove it.

(This project needs more tests. :-/)